spec/controllers/devise/two_factor_authentication_controller_spec.rb

```diff
@@ -71,4 +78,37 @@
       end
     end
   end
+
+  describe '#new' do
+    let(:user) { create(:user, :signed_up) }
+
+    context 'when not yet fully authenticated' do
+      before do
+        sign_in_before_2fa
+      end
+
+      it 'redirects to :show' do
+        get :new
+        expect(response).to redirect_to(action: :show)
```

Comment:
minor style nit: please add an empty line above the expectation to separate it from the rest of the test.